### PR TITLE
support geopoint as dimension

### PR DESCRIPTION
**Pull request by shz117**

### CODE REVIEW

File: query/common/dimval.go [hunk 1]
```diff
@@ -89,6 +89,14 @@ func ReadDimension(valueStart, nullStart unsafe.Pointer,
 			return &result
 		}
 		return nil
+	case memCom.GeoPoint:
+		coords := *(*[2]float32)(valuePtr)
```
Comment (lucafuji) [inline query/common/dimval.go]: somehow I think we can reuse the code in memcom.ConvertToHumanReadable

File: query/functor.hpp [hunk 2]
```diff
@@ -575,6 +579,86 @@ struct UnaryFunctor<UUIDT, UUIDT> {
...
+  typedef thrust::tuple<GeoPointT, bool> argument_type;
+  typedef thrust::tuple<O, bool> result_type;
+
+  explicit UnaryFunctor(UnaryFunctorType functorType)
+      : functorType(functorType) {}
+
+  UnaryFunctorType functorType;
+
+  __host__ __device__ result_type operator()(const argument_type t) const {
+       O o;
```
Comment (lucafuji) [inline query/functor.hpp]: Maybe a cleaner way to do this is disable binding non-geopoint output type in dimension_transform.cu

File: query/functor.hpp [hunk 3]
```diff
@@ -652,11 +736,34 @@ struct BinaryFunctor<UUIDT, I> {
   }
 };
 
+template <typename I>
```
Comment (lucafuji) [inline query/functor.hpp]: Why we allow binary functor for geo point?

Comment (shz117) [reply query/functor.hpp]: I don't know. I didn't want to allow it.
the code won't compile without these specializations.

File: query/functor.hpp (hunk 2 above)
Comment (shz117) [reply query/functor.hpp]: sorry I didn't get it. we do still want to support other types as dimensions.

**Pull request merged**